Add service_number to gitlab config defaults

mcelicalderon opened the pull request:
This PR adds `service_number` to Gitlab's default config, reads [CI_PIPELINE_ID](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html).
This allows to run [parallel builds](https://docs.coveralls.io/parallel-build-webhook) with Gitlab, current code only sets job ID, so no way to post a pipeline ID on the webhook as you get `{"error":"No build matching CI build number <JOB_ID> found"}` as [this build](https://gitlab.com/graphql-devise/graphql_devise/-/jobs/1097196689) shows.

https://github.com/graphql-devise/graphql_devise/pull/160 uses this fork and branch to run a parallel build successfully.

[here](https://gitlab.com/graphql-devise/graphql_devise/-/pipelines/270319900) is the build that works, and here is the [coveralls build](https://coveralls.io/builds/37920579).

mcelicalderon commented:
@afinetooth are you taking PRs on this gem? I don't see another way to support parallel builds on Gitlab, but maybe I'm missing something.

afinetooth commented:
@mcelicalderon until this can get reviewed, here is a public project using Gitlab CI to submit parallel builds to Coveralls:
https://gitlab.com/graphql-devise/graphql_devise

It is using the this gem as-is, so can perhaps be a model for you to follow if you're also using the gem in its current state. (Or are you using your own fork with your changes?)

afinetooth commented:
@mcelicalderon can you help me identify the "fork" and "branch" you mention here?:

> [graphql-devise/graphql_devise#160](https://github.com/graphql-devise/graphql_devise/pull/160) uses this fork and branch to run a parallel build successfully.

I'm also looking to add support for merge requests by adding [CI_MERGE_REQUEST_ID](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html#predefined-variables-for-merge-request-pipelines) / [CI_MERGE_REQUEST_IID](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html#predefined-variables-for-merge-request-pipelines).

I would like to test a project with support for both `service_number` and `service_pull_request` and get both merged asap.

afinetooth commented:
@mcelicalderon I found your fork:
https://github.com/graphql-devise/coveralls-ruby

And the relevant commit:
https://github.com/graphql-devise/coveralls-ruby/commit/2fe7b3568aae98325e33da94542902e99c4c0652

Thanks.

mcelicalderon commented:
Hey @afinetooth! Yes, I had to publish our own version of the gem so I could use it in graphql_devise. The original fork mentioned in this PR doesn't exist anymore as I moved the fork to the organization. I added something similar for CircleCI as I noticed it wouldn't work when using workflows and the build matrix, here's the commit https://github.com/graphql-devise/coveralls-ruby/commit/98f2b79e2aaff7cbece750fbccf4dc37a751b667

Here's the commit for the gitlab change https://github.com/graphql-devise/coveralls-ruby/commit/2fe7b3568aae98325e33da94542902e99c4c0652

I still have an open PR to move to CircleCI, so I might be able to provide more data on the ENV vars they are using. My changes on the fork are very specific to meet our needs, so not sure if the change is the right way to go for all scenarios. Happy to help if you think there's anything I can help with.

mcelicalderon commented:
@afinetooth closing this one in favor of #166 

This fork won't be removed
